Library/ViewModels/SettingsAccountViewModelTests.swift

```diff
@@ -8,65 +8,219 @@ import ReactiveSwift
-    let user = UserAccountFields.template
+    let user = GraphUser.template
       |> \.hasPassword .~ false
 
     let mockService = MockService(fetchGraphUserAccountFieldsResponse: UserEnvelope(me: user))
 
     withEnvironment(apiService: mockService) {
+      self.vm.inputs.viewDidLoad()
       self.vm.inputs.viewWillAppear()
-      self.reloadDataShouldHideWarningIcon.assertValueCount(1)
-      self.reloadDataCurrency.assertValueCount(1)
-      self.fetchAccountFieldsError.assertValueCount(0)
+      self.vm.inputs.viewDidAppear()
+
+      self.scheduler.advance()
+
+      self.reloadDataCurrency.assertValues([.USD])
+      self.reloadDataEmail.assertValues(["nativesquad@ksr.com"])
+      self.reloadDataIsAppleConnectedAccount.assertValues([false])
+      self.reloadDataShouldHideEmailPasswordSection
+        .assertValues([true], "Change email & password options are hidden.")
+      self.reloadDataShouldHideWarningIcon.assertValues([true])
+
+      self.fetchAccountFieldsError.assertDidNotEmitValue()
+
+      let shouldShowCreatePasswordFooter = self.vm.outputs.shouldShowCreatePasswordFooter()
+
+      XCTAssertEqual(true, shouldShowCreatePasswordFooter?.0)
+      XCTAssertEqual("nativesquad@ksr.com", shouldShowCreatePasswordFooter?.1)
     }
   }
 
   func testHideEmailPasswordHeaderView_HasPassword() {
-    let user = UserAccountFields.template
+    let user = GraphUser.template
+      |> \.hasPassword .~ true
+
+    let mockService = MockService(fetchGraphUserAccountFieldsResponse: UserEnvelope(me: user))
+
+    withEnvironment(apiService: mockService) {
+      self.vm.inputs.viewDidLoad()
+      self.vm.inputs.viewWillAppear()
+      self.vm.inputs.viewDidAppear()
+
+      self.scheduler.advance()
+
+      self.reloadDataCurrency.assertValues([.USD])
+      self.reloadDataEmail.assertValues(["nativesquad@ksr.com"])
+      self.reloadDataIsAppleConnectedAccount.assertValues([false])
+      self.reloadDataShouldHideEmailPasswordSection
+        .assertValues([false], "Change email & password options are shown if user has set a password")
+      self.reloadDataShouldHideWarningIcon.assertValues([true])
+
+      self.fetchAccountFieldsError.assertDidNotEmitValue()
+
+      let shouldShowCreatePasswordFooter = self.vm.outputs.shouldShowCreatePasswordFooter()
+
+      XCTAssertEqual(false, shouldShowCreatePasswordFooter?.0)
+      XCTAssertEqual("nativesquad@ksr.com", shouldShowCreatePasswordFooter?.1)
+    }
+  }
+
+  func testHideEmailWarningIcon_WhenEmailIsNotVerified_AndIsDeliverable() {
+    let user = GraphUser.template
       |> \.hasPassword .~ true
+      |> \.isEmailVerified .~ false
+      |> \.isDeliverable .~ true
 
     let mockService = MockService(fetchGraphUserAccountFieldsResponse: UserEnvelope(me: user))
 
     withEnvironment(apiService: mockService) {
+      self.vm.inputs.viewDidLoad()
       self.vm.inputs.viewWillAppear()
-      self.reloadDataShouldHideWarningIcon.assertValueCount(1)
```

Comment:
This was also not even testing what the value of `reloadDataShouldHideWarningIcon` was, which meant it could have been the opposite of what was expected and this test would have passed :\

Reply:
🙈